plugins/follow_me/follow_me_impl.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "follow_me.h"
+#include "mavlink_include.h"
+#include "plugin_impl_base.h"
+#include "device_impl.h"
+#include "timeout_handler.h"
+#include "global_include.h"
+
+namespace dronecore {
+
+class FollowMeImpl : public PluginImplBase
+{
+public:
+    FollowMeImpl();
+    ~FollowMeImpl();
+
+    void init() override;
+    void deinit() override;
+
+    const FollowMe::Config &get_config() const;
+    bool set_config(const FollowMe::Config &config);
+
+    FollowMe::Result start();
+    FollowMe::Result stop();
+
+    bool is_active() const;
+
+    FollowMe::Result set_curr_target_location(const FollowMe::TargetLocation &location);
+private:
+    void process_heartbeat(const mavlink_message_t &message);
+
+    // Config methods
+    void set_default_config();
+    bool is_config_ok(const FollowMe::Config &config) const;
+    void receive_param_min_height(bool success, float min_height_m);
+    void receive_param_follow_distance(bool success, float distance);
+    void receive_param_follow_direction(bool success, int32_t dir);
+    void receive_param_responsiveness(bool success, float rsp);
+    FollowMe::Result to_follow_me_result(MavlinkCommands::Result result) const;
+
+    void send_curr_target_location();
+    void stop_sending_target_location();
+
+    friend bool operator!(FollowMe::TargetLocation &location);
+
+    enum class EstimationCapabilites {
+        POS,
+        VEL
+    };
+
+    enum class Mode {
+        NOT_ACTIVE,
+        ACTIVE
+    } _mode = Mode::NOT_ACTIVE;
+    void set_mode(Mode mode);
+
+    mutable std::mutex _mutex {};
+    FollowMe::TargetLocation _curr_target_location {}; // sent to vehicle
+    void *_curr_target_location_cookie = nullptr;
+
+    Time _time;
```

Comment:
You can initialize most variables here like this:
```
Time _time {};
```
In my opinion it's encouraged over having a big constructor.